File: AppMovil-Kotlin/Backend/miapi/app/Models/Mercancia.php
<?php
// app/Models/Mercancia.php
// app/Models/Mercancia.php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Mercancia extends Model
{
    use HasFactory;

    protected $table = 'mercancias';

   

    protected $fillable = [
        'proveedor_id',
        'usuario_id',
        'fecha_ingreso',
        'num_remesa',
        'origen_mer',
        'destino_mer',
        'nom_remitente',
        'doc_remitente',
        'direccion_remitente',
        'cel_remitente',
        'destin_nom',
        'destin_doc',
        'destin_direccion',
        'destin_celular',
        'val_declarado',
        'val_flete',
        'val_total',
        'peso',
        'unidades',
        'observaciones',
        'tipopago_id'
    ];

    // Relaciones
    public function tipospago()
    {
        return $this->belongsTo(Tipopago::class, 'tipopago_id');
    }

    public function proveedores()
    {
        return $this->belongsTo(Proveedor::class, 'proveedor_id');
    }

    
}

File: src/main/kotlin/com/example/Sox/mercancias/Mercancia.kt
package com.example.Sox.mercancias

import java.time.LocalDate

data class Mercancia(
    val id: Int? = null,
    val proveedor_id: Int? = null,   // FK a proveedores
    val usuario_id: Int? = null,     // FK a usuarios
    val fecha_ingreso: LocalDate,
    val num_remesa: String? = null,
    val origen_mer: String,
    val destino_mer: String,

    val nom_remitente: String,
    val doc_remitente: String,
    val direccion_remitente: String,
    val cel_remitente: String,

    val destin_nom: String,
    val destin_doc: String,
    val destin_direccion: String,
    val destin_celular: String,

    val val_declarado: Double,
    val val_flete: Double,
    val val_total: Double,
    val peso: Double,
    val unidades: Int,
    val observaciones: String? = null,

    val tipopago_id: Int? = null
)
